### PR TITLE
Miron k2 agent oauth

### PR DESCRIPTION
**Pull request by mironabr**

### CODE REVIEW

File: src/main/java/com/k2view/agent/postman/CloudManager.java [hunk 1]
```diff
@@ -35,25 +36,50 @@ public CloudManager(String mailboxId, String mailboxUrl) {
...
-            String jsonArrayString = response.body();
+            final HttpResponse<String> response = client.postString(uri, body, null);
+            String jsonArrayString = response.body(); // Check response status code ?
             return Utils.gson.fromJson(jsonArrayString, Requests.class);
         } catch (Exception e) {
             Utils.logMessage("ERROR", "Failed to fetch messages from the server: " + e.getMessage());
             return new Requests(List.of(), 0);
         }
     }
+
```
Comment (idopolak) [inline src/main/java/com/k2view/agent/postman/CloudManager.java:71]: move to test

File: src/main/java/com/k2view/agent/postman/CloudManager.java [hunk 2]
```diff
@@ -35,25 +36,50 @@ public CloudManager(String mailboxId, String mailboxUrl) {
...
+     */
+
+    public static void main(String[] args) throws URISyntaxException {
+        final OAuthHttpSender.OAuthRequestBuilder oAuthBuilder =
+                OAuthHttpSender.newOAuthRequestBuilder("http://10.21.2.82:8081/sso-auth-server/oauth/token");
+        oAuthBuilder.clientId("bael-client").clientSecret("bael-secret").scope("1");
+        HttpSender client = oAuthBuilder.buildSender();
+        client.postString(new URI("http://google.com"),"hi",null);
+    }
 }
```
Comment (idopolak) [inline src/main/java/com/k2view/agent/postman/CloudManager.java:85]: where is it connected to the agent main?

**Review by idopolak — CHANGES_REQUESTED** [review]

File: src/main/java/com/k2view/agent/postman/CloudManager.java [hunk 3]
```diff
@@ -35,25 +36,50 @@ public CloudManager(String mailboxId, String mailboxUrl) {
         }
         this.mailboxId = mailboxId;
         this.uri = URI.create(mailboxUrl);
-        this.client = HttpClient.newBuilder().build();
+
```
Comment (idopolak) [inline src/main/java/com/k2view/agent/postman/CloudManager.java:39]: add env params to configure the different cases

Comment (mironabr) [reply src/main/java/com/k2view/agent/postman/CloudManager.java:39]: CloudManager.java is your code. Implement my code in your code however you want

File: src/main/java/com/k2view/agent/postman/CloudManager.java (hunk 1 above)
Comment (mironabr) [reply src/main/java/com/k2view/agent/postman/CloudManager.java:71]: CloudManager.java is your code. Implement my code in your code however you want

File: src/main/java/com/k2view/agent/postman/CloudManager.java (hunk 2 above)
Comment (mironabr) [reply src/main/java/com/k2view/agent/postman/CloudManager.java:85]: this is example/test code

**Review by idopolak — APPROVED** [review]

**Pull request merged**